s3/src/main/scala/quasar/destination/avalanche/s3/AvalancheS3DestinationModule.scala

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.avalanche.s3
+
+import argonaut._, Argonaut._
+import cats.data.EitherT
+import cats.effect.{Blocker, Concurrent, ConcurrentEffect, ContextShift, Resource, Sync, Timer}
+import cats.implicits._
+import doobie.hikari.HikariTransactor
+import eu.timepit.refined.auto._
+import java.util.concurrent.Executors
+import quasar.api.destination.DestinationError.InitializationError
+import quasar.api.destination.{DestinationError, DestinationType}
+import quasar.blobstore.s3.{
+  S3DeleteService,
+  S3PutService,
+  S3StatusService,
+  AccessKey,
+  Bucket,
+  Region,
+  SecretKey
+}
+import quasar.blobstore.BlobstoreStatus
+import quasar.connector.MonadResourceErr
+import quasar.connector.destination.{Destination, DestinationModule}
+import quasar.{concurrent => qc}
+
+import scala.{
+  Int,
+  StringContext,
+  Unit
+}
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.Predef.String
+import scala.util.{Either, Random}
+import scalaz.NonEmptyList
+import software.amazon.awssdk.services.s3.S3AsyncClient
+import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, StaticCredentialsProvider}
+import software.amazon.awssdk.regions.{Region => AwsRegion}
+
+object AvalancheS3DestinationModule extends DestinationModule {
```

Comment:
There's a LOT of stuff in here which is duplicated from the Azure module and can be factored into core.